Purchases/Purchasing/Purchases.swift

```diff
@@ -1160,6 +1160,43 @@ public extension Purchases {
     }
 #endif
 
+    /**
+     * Use this function to retrieve a `Bool` if eligible for a given `StoreProductDiscount` for a given `StoreProduct`.
```

Comment:
Could we utilize the existing [IntroEligibility](https://github.com/RevenueCat/purchases-ios/blob/main/Purchases/Purchasing/IntroEligibility.swift) enum?

Reply:
I initially made it as a `Bool` so that we didn't have to make a new public type that also worked in Objective-C 🤷‍♂️ Are there are states besides eligible and not eligible where an enum would make more sense?

Reply:
Eh, probably not - I think we'll either verify the discount or not, so it's probably safe. Just wanted to ask 😄

Reply:
I appreciate the question because I didn't know about the `IntroElegibility` enum so I learned something new 😊

Reply:
I do like the consistency of reusing the enum

Reply:
@aboedo I would vote to rename the enum then because I think this would add to the conduction between "Intro Eligibility" and "Promotional Offer Eligibility" 🤔 

I also think using an enum as a user is less ideal than a boolean if we can use a boolean. Enum forces a switch or if case (which is kind of gross) and switches don't work in SwiftUI view builders yet (I don't believe 🤷‍♂️)

I'm okay with whichever you'd prefer just wanted to leave my thoughts here! 😊

Reply:
great points! I hadn't thought of that. The one advantage of an `enum`, though, is that if new cases do come up, we don't need a new major version to add them, we can be a lot more flexible (like for the case of intro eligibility, where we realized now that we need to add a new enum case for when an intro offer doesn't exist in the first place)

Reply:
@aboedo Ahhh, that is also a good point! 😱 And... there is a good chance that new cases will probably come up 😬 

Sooooooo... maybe we do make this an enum but I think it should probably be a new one one because reusing `IntroEligibility` would be confusing 😬

Maybe like `PromotionalCodeEligibility`? Thoughts?

Reply:
[PromoCodes are their own thing](https://docs.revenuecat.com/docs/ios-subscription-offers#types-of-subscription-offers), how about `PromotionalOfferEligibility`?

Reply:
Omg, I got the thing things confused already 🤦‍♂️ Why why why...

But yes, you right!